### PR TITLE
nil reference and request type fix

### PR DESCRIPTION
**Pull request by kuldeep0020**
Changed /delete-users API call type to POST from DELETE
Initialised nil objects wherever necessary

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: regulation-worker/cmd/main.go [hunk 1]
```diff
@@ -34,12 +35,16 @@ func main() {
...
+	apiManager := api.APIManager{
+		Client:           &http.Client{},
+		DestTransformURL: transformerURL,
+	}
+	router := delete.NewRouter(&kvstore.KVDeleteManager{}, &batch.BatchManager{}, &apiManager)
 	svc := service.JobSvc{
 		API: &client.JobAPI{
 			WorkspaceID: config.GetEnv("workspaceID", "1001"),
-			URLPrefix:   config.GetEnv("urlPrefix", "https://api.rudderlabs.com:35359"),
+			URLPrefix:   config.GetEnv("urlPrefix", "http://localhost:35359"),
```
Comment (lvrach) [inline regulation-worker/cmd/main.go:47]: Does `http://localhost:35359` make sense as default?

**Review by lvrach — APPROVED** [review]

File: regulation-worker/cmd/main.go (hunk 1 above)
Comment (kuldeep0020) [reply regulation-worker/cmd/main.go:47]: Changing it.

File: regulation-worker/internal/service/service.go [hunk 2]
```diff
@@ -30,9 +30,11 @@ type JobSvc struct {
...
 }
 
-//called by looper
-//calls api-client.getJob(workspaceID)
-//calls api-client to get new job with workspaceID, which returns jobID.
+/* called by looper
+calls api-client.getJob(workspaceID)
+ calls api-client to get new job with workspaceID, which returns jobID.
+ */
+
```
Comment (saurav-malani) [inline regulation-worker/internal/service/service.go:37]: [Minor]
Line comments are the norm; block comments appear mostly as package comments. ([Ref](https://go.dev/doc/effective_go#commentary))

File: regulation-worker/internal/service/service.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ type JobSvc struct {
 //called by looper
 //calls api-client.getJob(workspaceID)
 //calls api-client to get new job with workspaceID, which returns jobID.
+
```
Comment (kuldeep0020) [inline regulation-worker/internal/service/service.go:36]:
```suggestion
```

**Review by saurav-malani — APPROVED** [review]

**Review by chandumlg — APPROVED** [review]

**Pull request merged**